spring-app/src/main/java/pl/edu/agh/cqm/service/ThroughputServiceImpl.java

```diff
@@ -21,25 +22,24 @@
 
     private final PcapNetworkInterface.PromiscuousMode mode = PcapNetworkInterface.PromiscuousMode.NONPROMISCUOUS;
     private final ThroughputSampleRepository dataRepository;
-    private final ParameterService parameterService;
-    private final CqmConfiguration configuration;
-    private final Logger logger = LogManager.getLogger(ThroughputServiceImpl.class);
+    private final Logger logger;
     private final int snapLen;
     private final int timeout;
+    private final ParameterService parameterService;
     private int measurementTime;
     private final int sessionBreakTime;
     private final String interfaceName;
     private String myIP;
     private PcapNetworkInterface nif;
+    private List<CDNsData> cdns;
 
-    public ThroughputServiceImpl(CqmConfiguration configuration,
-                                 ThroughputSampleRepository dataRepository,
-                                 ParameterService parameterService) throws PcapNativeException {
-        this.configuration = configuration;
+    public ThroughputServiceImpl(CqmConfiguration configuration,ParameterService parameterService, ThroughputSampleRepository dataRepository) throws PcapNativeException {
+        logger = LogManager.getLogger("ThroughputServiceImpl");
         this.dataRepository = dataRepository;
         this.parameterService = parameterService;
         snapLen = configuration.getPcapMaxPacketLength();
         timeout = configuration.getPcapTimeout();
+        measurementTime = 1000 * 60 * parameterService.getPassiveSamplingRate();
```

Comment:
`measurementTime` has to be updated before every measurement (in case it gets updated). That's why I put it inside `doMeasurement` in my PR.